### PR TITLE
mpls-conf: T915: Add params cisco-tlv and prefer ipv4 transport

### PR DESCRIPTION
**Pull request by sever-sever**
Add new commands for mpls ldp
```
set protocols mpls ldp cisco-interop-tlv
set protocols mpls ldp discovery transport-prefer-ipv4
```

### CODE REVIEW

**Comment by c-po** [comment]
I have no experience in operating an MPLS backbone, but wondering if the CLI should not match something more to what we already got for OSPF and BGP:

```
set protocols mpls parameters ldp-dual-stack 'cisco'
set protocols mpls parameters ldp-transport-prefer-ipv4
```

But I might be on an entire wrong wrong so please proof me wrong. From an implementation point of view it LGTM.

**Comment by sever-sever** [comment]
I have no experience with mpls to.
I can make the "parameters" option.

**Comment by Cheeze-It** [comment]
Per talking to c-po, on miscellaneous features the conversation basically was agreed upon that we use for all protocols a node called "parameters" and just to stick everything else in there that doesn't really fit anywhere. 

So using the above of would be completely ok:

set protocols mpls parameters ldp-dual-stack 'cisco'
set protocols mpls parameters ldp-transport-prefer-ipv4

I'll work with Viacheslav more on the other options too. 

If need be I can research exactly what those options do and add to the help and descriptions so that they are accurate.

**Comment by sever-sever** [comment]
Updated PR with section "parameters"

```
set protocols mpls ldp parameters cisco-interop-tlv
set protocols mpls ldp parameters transport-prefer-ipv4

```

File: interface-definitions/protocols-mpls.xml.in [hunk 1]
```diff
@@ -14,6 +14,25 @@
...
+                <children>
+                  <leafNode name="cisco-interop-tlv">
+                    <properties>
+                      <help>Enable Cisco non-compliant format capability TLV</help>
+                      <valueless/>
+                    </properties>
+                  </leafNode>
+                  <leafNode name="transport-prefer-ipv4">
+                    <properties>
+                      <help>Preferred IPv4 for TCP peer transport connection</help>
```
Comment (dmbaturin) [inline interface-definitions/protocols-mpls.xml.in]: This is a bit of a nitpicking, but let's make help string consistent. The unwritten policy is to use verbs, so this should be "Prefer IPv4..."

**Review by dmbaturin — CHANGES_REQUESTED** [review]
Otherwise I see no problem with this implementation.

File: interface-definitions/protocols-mpls.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/protocols-mpls.xml.in]: Done

**Comment by c-po** [comment]
Conflicting files
src/conf_mode/protocols_mpls.py

**Comment by Cheeze-It** [comment]
> Conflicting files
> src/conf_mode/protocols_mpls.py

Ahh I see what's going on. Because mine was merged and it affected src/conf_mode/protocols_mpls.py then adding this PR would remove my changes in https://github.com/vyos/vyos-1x/pull/585 right?

**Comment by c-po** [comment]
It‘s wat Github sees. Please do a git pull —rebase from this repository and force push after resolving the conflicts.

**Comment by sever-sever** [comment]
I can't figure it out. So I close that PR.

```
vyos_bld@927195d8138b:/vyos/work/T915_cisco/vyos-1x$ git pull --rebase
remote: Enumerating objects: 419, done.
remote: Counting objects: 100% (419/419), done.
remote: Compressing objects: 100% (54/54), done.
remote: Total 544 (delta 372), reused 408 (delta 365), pack-reused 125
Receiving objects: 100% (544/544), 115.06 KiB | 973.00 KiB/s, done.
Resolving deltas: 100% (394/394), completed with 155 local objects.
From https://github.com/sever-sever/vyos-1x
 * [new branch]        T1117                  -> origin/T1117
 * [new branch]        T1316_october          -> origin/T1316_october
 * [new branch]        T2103                  -> origin/T2103
 * [new branch]        T2277                  -> origin/T2277
 * [new branch]        T2387                  -> origin/T2387
 * [new branch]        T2512-fix              -> origin/T2512-fix
 * [new branch]        T2547                  -> origin/T2547
 * [new branch]        T2547-rip              -> origin/T2547-rip
 * [new branch]        T2578                  -> origin/T2578
 * [new branch]        T2617                  -> origin/T2617
 * [new branch]        T2631                  -> origin/T2631
 * [new branch]        T2658                  -> origin/T2658
 * [new branch]        T2718                  -> origin/T2718
 * [new branch]        T2779                  -> origin/T2779
 * [new branch]        T2782                  -> origin/T2782
 * [new branch]        T2791                  -> origin/T2791
 * [new branch]        T2833                  -> origin/T2833
 * [new branch]        T2841                  -> origin/T2841
 * [new branch]        T2846                  -> origin/T2846
 * [new branch]        T2850                  -> origin/T2850
 * [new branch]        T2856                  -> origin/T2856
 * [new branch]        T2856_crux             -> origin/T2856_crux
 * [new branch]        T2874                  -> origin/T2874
 * [new branch]        T2915                  -> origin/T2915
 * [new branch]        T2917                  -> origin/T2917
 * [new branch]        T2917_port             -> origin/T2917_port
 * [new branch]        T2918                  -> origin/T2918
 * [new branch]        T2933                  -> origin/T2933
 * [new branch]        T2933_crux             -> origin/T2933_crux
 * [new branch]        T2938                  -> origin/T2938
 * [new branch]        T2944                  -> origin/T2944
 * [new branch]        T2965                  -> origin/T2965
 * [new branch]        T2982_crux             -> origin/T2982_crux
 * [new branch]        T3031                  -> origin/T3031
 * [new branch]        T752                   -> origin/T752
 * [new branch]        T915-timers            -> origin/T915-timers
 * [new branch]        T915_holdtime_explicit -> origin/T915_holdtime_explicit
 * [new branch]        T915_ttl               -> origin/T915_ttl
 * [new branch]        bgp-xml-conf           -> origin/bgp-xml-conf
 * [new branch]        bgp_first_steps        -> origin/bgp_first_steps
 * [new branch]        bmon                   -> origin/bmon
 * [new branch]        clear-igmp             -> origin/clear-igmp
 + 9c831496...e2ec9af4 crux                   -> origin/crux  (forced update)
 + 75b07725...807b8871 current                -> origin/current  (forced update)
 * [new branch]        desc-typo              -> origin/desc-typo
 * [new branch]        dns-for-stat           -> origin/dns-for-stat
 * [new branch]        eigrp-conf-mode        -> origin/eigrp-conf-mode
 * [new branch]        fast                   -> origin/fast
 * [new branch]        isis_test              -> origin/isis_test
 * [new branch]        new-init               -> origin/new-init
 * [new branch]        new-pppoe              -> origin/new-pppoe
 * [new branch]        op-pim-igmp            -> origin/op-pim-igmp
 * [new branch]        op-show-bgp            -> origin/op-show-bgp
 * [new branch]        policy-prefix          -> origin/policy-prefix
 * [new branch]        xml-conf-rip           -> origin/xml-conf-rip
 * [new branch]        xml-isis               -> origin/xml-isis
There is no tracking information for the current branch.
Please specify which branch you want to rebase against.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> T915_cis

vyos_bld@927195d8138b:/vyos/work/T915_cisco/vyos-1x$ git status
On branch T915_cis
nothing to commit, working tree clean
vyos_bld@927195d8138b:/vyos/work/T915_cisco/vyos-1x$ 
vyos_bld@927195d8138b:/vyos/work/T915_cisco/vyos-1x$ git push origin T915_cis --force
Everything up-to-date

```

@Cheeze-It add please new one if you really need it.

**Comment by Cheeze-It** [comment]
@sever-sever, yessir, I'll see if I can get the PR going. 

Thank you guys.

**Pull request closed without merging**